.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,40 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - '**'
```

Comment:
We can exclude the `main` branch because the other workflow will be executed on merge to `main` branch.